src/EditorFeatures/Core/Wrapping/ChainedExpression/AbstractChainedExpressionWrapper.cs

```diff
@@ -0,0 +1,309 @@
+            var pieces = ArrayBuilder<SyntaxNodeOrToken>.GetInstance();
+            Decompose(node, pieces);
+
+            // Now that we have the pieces, find 'chunks' similar to the form:
+            //
+            //      . Name (...) Remainder...
+            //
+            // These will be the chunks that are wrapped and aligned on that dot.
+            // 
+            // Here 'remainder' is everything up to the next `. Name (...)` chunk.
+
+            var chunks = ArrayBuilder<ImmutableArray<SyntaxNodeOrToken>>.GetInstance();
+            BreakPiecesIntoChunks(pieces, chunks);
+
+            pieces.Free();
+            return chunks.ToImmutableAndFree();
+        }
+
+        private void BreakPiecesIntoChunks(
+            ArrayBuilder<SyntaxNodeOrToken> pieces,
+            ArrayBuilder<ImmutableArray<SyntaxNodeOrToken>> chunks)
+        {
+            // Have to look for the first chunk after the first piece.  i.e. if the pieces
+            // starts with `.Foo().Bar().Baz()` then the chunks would be `.Bar()` and `.Baz()`.
+            // However, if we had `this.Foo().Bar().Baz()` then the chunks would be `.Foo()`
+            // `.Bar()` and `.Baz()`.
+            //
+            // Note: the only way to get the `.Foo().Bar().Baz()` case today is in VB in
+            // a 'with' statement.  if we have that, we don't want to wrap it into:
+            //
+            //  <c>
+            //  with ...
+            //      .Foo()
+            //      .Bar()
+            //      .Baz()
+            //  <c>
+            //
+            // Instead, we want to create
+            //
+            //  <c>
+            //  with ...
+            //      .Foo().Bar()
+            //            .Baz()
+            //  <c>
+            var currentChunkStart = FindNextChunkStart(pieces, firstChunk: true, index: 1);
+            if (currentChunkStart < 0)
+            {
+                return;
+            }
+
+            while (true)
+            {
+                // Look for the next chunk starting after the current chunk we're on.
+                var nextChunkStart = FindNextChunkStart(pieces, firstChunk: false, index: currentChunkStart + 1);
+                if (nextChunkStart < 0)
+                {
+                    // No next chunk after the current one.  The current chunk just
+                    // extends to the end of the pieces.
+                    chunks.Add(GetSubRange(pieces, currentChunkStart, end: pieces.Count));
+                    return;
+                }
+
+                // Had a chunk after this one.  Record the current chunk, move to the start
+                // of the next one, and then keep going.
+                chunks.Add(GetSubRange(pieces, currentChunkStart, end: nextChunkStart));
+                currentChunkStart = nextChunkStart;
+            }
+        }
+
+        /// <summary>
+        /// Looks for the next sequence of `. Name (ArgList)`.  Note, except for the first
```

Comment:
❕ `` ` `` is not a supported formatting character in XML documentation comments. Wrap code with `<c>`...`</c>` instead.

```suggestion
        /// Looks for the next sequence of <c>. Name (ArgList)</c>.  Note, except for the first
```

Reply:
Fixed everywhere.